src/components/TopBar/AboutDialog.js

```diff
@@ -5,107 +5,127 @@
-    other: <QuestionMark />,
+const ComponentsTypesAvatar = {
+    app: <WidgetsOutlined fontSize="small" color="primary" />,
+    server: <DnsOutlined fontSize="small" color="secondary" />,
+    other: <QuestionMark fontSize="small" />,
 };
 
+const typeSort = {
+    app: 1,
+    server: 10,
+    other: 20,
+};
+function compareAdditionalComponents(c1, c2) {
+    //sort by type then by name
+    return (
+        [typeSort[c1.type] || 100] - [typeSort[c2.type] || 100] ||
+        (c1.name || '').localeCompare(c2.name || '')
+    );
+}
+
 const AboutDialog = ({
     open,
     onClose,
     getGlobalVersion,
-    getLogoThemed,
+    appName,
     appVersion,
+    appGitTag,
     appLicense,
     getAdditionalComponents,
 }) => {
     const theme = useTheme();
-
-    //TODO is useCallback in component or in caller?
-    const logo = useMemo(() => {
-        if (getLogoThemed) {
-            return getLogoThemed(theme.palette.mode);
-        } else {
-            return <BrokenImage />;
-        }
-    }, [getLogoThemed, theme.palette.mode]);
-
-    const handlerGetGlobalVersion = useCallback(getGlobalVersion, [
-        getGlobalVersion,
-    ]);
+    const [isRefreshing, setRefreshState] = useState(false);
     const [loadingGlobalVersion, setLoadingGlobalVersion] = useState(false);
 
     /* We want to get the initial version once at first render to detect later a new deploy */
-    const [startingGlobalVersion, setStartingGlobalVersion] = useState(null);
+    const [startingGlobalVersion, setStartingGlobalVersion] =
+        useState(undefined);
     useEffect(() => {
-        if (getGlobalVersion) {
-            handlerGetGlobalVersion((value) => {
-                setLoadingGlobalVersion(false);
+        if (startingGlobalVersion === undefined && getGlobalVersion) {
+            getGlobalVersion((value) => {
                 setStartingGlobalVersion(value);
                 setActualGlobalVersion(value);
             });
         }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, []);
+    }, [getGlobalVersion, startingGlobalVersion]);
 
     const [actualGlobalVersion, setActualGlobalVersion] = useState(null);
     useEffect(() => {
         if (open && getGlobalVersion) {
             setLoadingGlobalVersion(true);
-            handlerGetGlobalVersion((value) => {
+            getGlobalVersion((value) => {
                 setLoadingGlobalVersion(false);
-                setActualGlobalVersion(value);
+                setActualGlobalVersion(value || null);
             });
         }
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [open, handlerGetGlobalVersion]);
+    }, [open, getGlobalVersion]);
 
+    const [openAdditionalComponents, setOpenAdditionalComponents] =
+        useState(true);
     const [loadingAdditionalComponents, setLoadingAdditionalComponents] =
         useState(false);
     const [additionalComponents, setAdditionalComponents] = useState(null);
```

Comment:
If the current app is inside this array, then it's not `additionalComponents` but just `components` or `allComponents` no ?